diff -rupN swaylock-1.8.0-orig/shadow.c swaylock-1.8.0/shadow.c
--- swaylock-1.8.0-orig/shadow.c	2024-08-24 14:37:15.125772418 +0300
+++ swaylock-1.8.0/shadow.c	2024-08-24 14:38:10.253743735 +0300
@@ -34,6 +34,7 @@ void initialize_pw_backend(int argc, cha
 		encpw = swent->sp_pwdp;
 	}
 
+#if 0
 	if (setgid(getgid()) != 0) {
 		swaylock_log_errno(LOG_ERROR, "Unable to drop root");
 		exit(EXIT_FAILURE);
@@ -47,6 +48,7 @@ void initialize_pw_backend(int argc, cha
 			"able to restore it after setuid/setgid)");
 		exit(EXIT_FAILURE);
 	}
+#endif
 
 	/* This code does not run as root */
 	swaylock_log(LOG_DEBUG, "Prepared to authorize user %s", pwent->pw_name);
