diff -rupN gtkdialog-0.8.5a-orig/src/actions.c gtkdialog-0.8.5a/src/actions.c
--- gtkdialog-0.8.5a-orig/src/actions.c	2023-08-03 20:07:12.760221783 +0300
+++ gtkdialog-0.8.5a/src/actions.c	2023-08-03 20:07:50.729226584 +0300
@@ -642,18 +642,34 @@ void action_presentwindow(GtkWidget *wid
 /***********************************************************************
  * Action command                                                      *
  ***********************************************************************/
+
+static void _action_shellcommand(const char *command)
+{
+        char *argv[] = {"bash", "-c", command, NULL};
+ 
+        g_spawn_sync(NULL,
+                     argv,
+                     NULL,
+                     G_SPAWN_LEAVE_DESCRIPTORS_OPEN | G_SPAWN_SEARCH_PATH,
+                     NULL,
+                     NULL,
+                     NULL,
+                     NULL,
+                     NULL,
+                     NULL);
+}
+
 /* This fuction will export variables and run a shell command */
 
 void action_shellcommand(GtkWidget *widget, char *string)
 {
 	char *command;
-	int result;
 
 	variables_export_all();
 
 	if (option_include_file == NULL) {
 
-		result = system(string);
+		_action_shellcommand(string);
 
 	} else {
 
@@ -661,7 +677,7 @@ void action_shellcommand(GtkWidget *widg
 		command = g_strdup_printf("source %s; %s", */
 		command = g_strdup_printf(". %s; %s",
 				option_include_file, string);
-		result = system(command);
+		_action_shellcommand(command);
 		g_free(command);
 
 	}
