From 66730fd552434482402e3687c509490ef88c813d Mon Sep 17 00:00:00 2001
From: Dima Krasner <dima@dimakrasner.com>
Date: Fri, 9 Aug 2024 12:27:35 +0300
Subject: [PATCH] give keyboard focus when placed in the overlay layer

---
 src/main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/main.c b/src/main.c
index 86469ad..135754c 100644
--- a/src/main.c
+++ b/src/main.c
@@ -457,6 +457,10 @@ create_dialog (void)
   gtk_window_set_skip_taskbar_hint (GTK_WINDOW (dlg), options.data.skip_taskbar);
   gtk_window_set_skip_pager_hint (GTK_WINDOW (dlg), options.data.skip_taskbar);
   gtk_window_set_accept_focus (GTK_WINDOW (dlg), options.data.focus);
+#ifdef HAVE_GTK_LAYER_SHELL
+  if (options.data.focus && (layer != GTK_LAYER_SHELL_LAYER_ENTRY_NUMBER || edge != GTK_LAYER_SHELL_EDGE_ENTRY_NUMBER || corner != GTK_LAYER_SHELL_EDGE_ENTRY_NUMBER))
+    gtk_layer_set_keyboard_mode (GTK_WINDOW(dlg), GTK_LAYER_SHELL_KEYBOARD_MODE_EXCLUSIVE);
+#endif
 
   /* create box */
 #if !GTK_CHECK_VERSION(3,0,0)
-- 
2.43.0

